opm/common/utility/Serializer.hpp

```diff
@@ -86,20 +91,26 @@ class Serializer {
         std::memcpy(&this->buffer[write_pos], ptr, value_size);
     }
 
-    std::size_t pos = 0;
+    std::size_t read_pos = 0;
 };
 
 template <>
 void inline Serializer::put(const std::string& value) {
-    this->put<std::string::size_type>(value.size());
+    this->put(value.size());
+    if (value.empty())
+        return;
+
     this->pack(value.c_str(), value.size());
 }
 
 template<>
 std::string inline Serializer::get<std::string>() {
     std::string::size_type length = this->get<std::string::size_type>();
-    this->pos += length;
-    return {std::addressof(this->buffer[this->pos - length]), length};
+    if (length == 0)
+        return std::string{};
+
+    this->read_pos += length;
+    return {std::addressof(this->buffer[this->read_pos - length]), length};
```

Comment:
Mmh, are we sure that this works in the general case? This calls the constructor with the `const char*` parameter and hence  a C string. IMHO a c string should be `\0` delimited, but it seems like we never write the delimiter. Wouldn't we need to use:
```
this->pack(value.c_str(), value.size()+1);
```
above to also write that zero?

Reply:
Oversaw the length argument.